### PR TITLE
[Merged by Bors] - POET: do not re-validate a existing proof

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation

skip expensive validation for existing proof

## Changes

- add a sql existence lookup by the proof reference
- ignore the p2p msg if the proof already exists 
- skip validating PoET proof with empty membership

## Test Plan

unit test

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: activation/poetdb.go [hunk 1]
```diff
@@ -34,18 +34,27 @@ func NewPoetDb(db *sql.Database, log log.Log) *PoetDb {
...
-	return err == nil
+	has, err := poets.Has(db.sqlDB, proofRef)
+	return err == nil && has
 }
 
 // ValidateAndStore validates and stores a new PoET proof.
 func (db *PoetDb) ValidateAndStore(proofMessage *types.PoetProofMessage) error {
+	ref, err := proofMessage.Ref()
+	if err != nil {
+		return err
```
Comment (evt) [inline activation/poetdb.go:45]: @countvonzero do we skip `fmt.Errorf` here on purpose?

Comment (countvonzero) [reply activation/poetdb.go:45]: more like a Freudian slip on my part. 
not a fan of suing fmt.Errorf, especially when the code stack is deep. just make the error unreadable.
prefer to use context for debugging instead.

Comment (evt) [reply activation/poetdb.go:45]: Interesting point :) On the contrary I try to wrap as much as possible everywhere as then you see the whole execution stack in one line and don't need anything else to find the issue :)

File: activation/poetdb.go [hunk 2]
```diff
@@ -64,12 +73,16 @@ func (db *PoetDb) Validate(proof types.PoetProof, poetID []byte, roundID string,
 		return types.ProcessingError(fmt.Sprintf("invalid poet id %x", poetID))
 	}
 	root, err := calcRoot(proof.Members)
+	// we shouldn't care about poet proof with empty membership as it's not relevant.
+	if len(proof.Members) == 0 {
```
Comment (countvonzero) [inline activation/poetdb.go:77]: can you double check we no longer has this log with this check?
```
{"L":"ERROR","T":"2022-04-15T08:23:53.242-0400","N":"35daa.poetListener ","M":"failed to store poet proof: failed to store poet proof for poetId 09146d5a32 round 4: exec: database: object exists","node_id":"35daab80af8a9960524a940cad5f8ab769d93afb483d618199da65b2ba5cf87a","module":"poetListener"}
```

Comment (evt) [reply activation/poetdb.go:77]: @countvonzero do you remember which systest was it?

Comment (countvonzero) [reply activation/poetdb.go:77]: any systest. this is part of nodes setup.

Comment (evt) [reply activation/poetdb.go:77]: I've checked error fixed:

before (error is there) - http://35.223.10.224/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22Loki%22,%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22test-icul%5C%22%7D%7C%20json%20%7C%3D%20%5C%22failed%20to%20store%20poet%20proof%5C%22%22%7D%5D

after (no error) - http://35.223.10.224/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22Loki%22,%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22test-ndat%5C%22%7D%7C%20json%20%7C%3D%20%5C%22failed%20to%20store%20poet%20proof%5C%22%22%7D%5D

Comment (countvonzero) [reply activation/poetdb.go:77]: cannot log in the provided link. can you provide screen shots instead? thanks

Comment (evt) [reply activation/poetdb.go:77]: I've sent you a password in Slack. Let me know if screenshots still needed.

Comment (evt) [reply activation/poetdb.go:77]: sure, here they are
Screenshot 2022-05-04 at 13 05 01
Screenshot 2022-05-04 at 13 05 12

**Review by countvonzero — COMMENTED** [review]
@dshulyak this PR LGTM but i cannot approve. need your review. thanks

**Review by dshulyak — APPROVED** [review]

**Comment by evt** [comment]
bors try

**Comment by evt** [comment]
bors merge

**Pull request closed without merging**